### DISCUSSION TITLE
AuthnContextClassRef always seems to be 'unspecified' for Keycloak SAML IDP authentication

### DISCUSSION THREAD

**Discussion by jeffharwell** [keycloak/keycloak#11688] [Q&A]
Greetings,

If anyone has any insight on this I would greatly appreciate it. 

We are on Keycloak 15.0.2 and I am connecting Keycloak as a SAML IDP to an ADFS SP. ADFS is requesting a specific AuthnContextClassRef in the SAML AuthnRequest:

```
    <samlp:RequestedAuthnContext>
        <samlp:AuthnContextClassRef xmlns:samlp="urn:oasis:names:tc:SAML:2.0:assertion">urn:oasis:names:tc:SAML:2.0:ac:classes:Password</samlp:AuthnContextClassRef>
    </samlp:RequestedAuthnContext>
```
The client defined in Keycloak is using the 'browser' based authentication flow which uses password authentication. After successful authentication on Keycloak using a username/password the SAML response Keycloak generates shows an AuthnContextClassRef of "unspecified" as below:

```
            <saml:AuthnContext>
                <saml:AuthnContextClassRef>urn:oasis:names:tc:SAML:2.0:ac:classes:unspecified</saml:AuthnContextClassRef>
            </saml:AuthnContext>
        </saml:AuthnStatement>
```
How can I get Keycloak to to switch this to Password? Just to be clear, I'm not using Keycloak as a broker, this is just the default keycloak identity provider.

Any pointers would be greatly appreciated. Thank you for reading.

Jeff

**Comment by jeffharwell** [comment]
I just install version 18 and confirmed that I am seeing the same behavior.

**Comment by pselvini** [comment]
Same issue here. Jeff, did you manage to solve this?

thanks
Paolo
